internal/core/algorithm/ngt/option.go

```diff
@@ -51,38 +52,47 @@ var (
 	}
 )
 
+// WithInMemoryMode represents the option to set to start in memory mode or not for NGT.
 func WithInMemoryMode(flg bool) Option {
 	return func(n *ngt) error {
 		n.inMemory = flg
 		return nil
 	}
 }
 
+// WithIndexPath represents the option to set the index path for NGT.
 func WithIndexPath(path string) Option {
 	return func(n *ngt) error {
 		if len(path) == 0 {
-			return nil
+			return errors.NewErrInvalidOption("indexPath", path)
```

Comment:
I think running agents without `indexPath` is normal usage.
Does it show a warn error in log?
If it does, i think it is better to create a new level of option error like `ErrUnsetOption` (maybe there're better names) and it causes only a debug log.

Reply:
Yes, it shows the warn log in agent.
@vankichi @hlts2 what do you think?

Reply:
thank you for comment.

>  i think it is better to create a new level of option error like ErrUnsetOption (maybe there're better names) and it causes only a debug log.

I understood it. and I think it looks good to me.
What about the `IgnoreOption` for the option error name?

Reply:
@hlts2 I have fixed rinx comment, can you please review it again?

Reply:
@hlts2 Sorry for late comment.
I think `IgnoreOption` like the option is being ignored, which is not (we will print the debug log), so I think `ErrUnsetOption` is better for me.
What do you think?